package/MDAnalysis/analysis/dssp/pydssp_numpy.py

```diff
@@ -0,0 +1,276 @@
+    unfolded = np.take(a, idx, axis=axis)
+    return np.moveaxis(unfolded, axis - 1, -1)
+
+
+def _get_hydrogen_atom_position(coord: np.ndarray) -> np.ndarray:
+    """Fills in hydrogen atoms positions if they are abscent, under the
+    assumption that C-N-H and H-N-CA angles are perfect 120 degrees,
+    and N-H bond length is 1.01 A.
+
+    Parameters
+    ----------
+    coord : np.ndarray
+        input coordinates in Angstrom, shape (n_atoms, 4, 3),
+        where second axes corresponds to (N, CA, C, O) atom coordinates
+
+    Returns
+    -------
+    np.ndarray
+        coordinates of additional hydrogens, shape (n_atoms-1, 3)
+
+    .. versionadded:: 2.8.0
+    """
+    # C_i, N_i, H_i and CA_{i+1} are all in the peptide bond plane
+    # we wanna get C_{i+1} - N_{i} vectors and normalize them
+    # ---------
+    # v1 = vec(C_i, N_i)
+    # v2 = vec(CA_{i+1}, N_i)
+    # v3 = vec(N_i, H_i) = ?
+    # we use the assumption that all the angles are 120 degrees,
+    # and |v3| = 1.01, hence
+    # we can derive v3 = (v1/|v1| + v2/|v2|)*|v3|
+
+    # get v1 = vec(C_i, N_i)
+    vec_cn = coord[1:, 0] - coord[:-1, 2]
+    vec_cn = vec_cn / np.linalg.norm(vec_cn, axis=-1, keepdims=True)
+
+    # get v2 = vec(CA_{i+1}, N_{i})
+    vec_can = coord[1:, 0] - coord[1:, 1]
+    vec_can = vec_can / np.linalg.norm(vec_can, axis=-1, keepdims=True)
+
+    vec_nh = vec_cn + vec_can
+    vec_nh = vec_nh / np.linalg.norm(vec_nh, axis=-1, keepdims=True)
+
+    # vec_(0, H) = vec(0, N) + vec_nh
+    return coord[1:, 0] + 1.01 * vec_nh
+
+
+def get_hbond_map(
+    coord: np.ndarray,
+    cutoff: float = DEFAULT_CUTOFF,
+    margin: float = DEFAULT_MARGIN,
+    return_e: bool = False,
+) -> np.ndarray:
+    """Returns hydrogen bond map
+
+    Parameters
+    ----------
+    coord : np.ndarray
+        input coordinates in either (n, 4, 3) or (n, 5, 3) shape
+        (without or with hydrogens). If hydrogens are not present, then
+        ideal positions (see :func:_get_hydrogen_atom_positions) are used.
+    cutoff : float, optional
+        cutoff, by default DEFAULT_CUTOFF
+    margin : float, optional
+        margin, by default DEFAULT_MARGIN
+    return_e : bool, optional
+        if to return energy instead of hbond map, by default False
+
+    Returns
+    -------
+    np.ndarray
+        output hbond map or energy depending on return_e param
+
+    Raises
+    ------
+
+    ImportError
+        if module `einops` is not present
+
+    .. versionadded:: 2.8.0
+    """
+    n_atoms, n_atom_types, _ = coord.shape
+    assert n_atom_types in (
+        4,
+        5,
+    ), "Number of atoms should be 4 (N,CA,C,O) or 5 (N,CA,C,O,H)"
+
+    if n_atom_types == 4:
+        h_1 = _get_hydrogen_atom_position(coord)
+    elif n_atom_types == 5:
+        h_1 = coord[1:, 4]
+        coord = coord[:, :4]
+    else:
+        raise ValueError("Number of atoms should be 4 (N,CA,C,O) or 5 (N,CA,C,O,H)")
```

Comment:
explicitly mark block as not tested

Reply:
I marked the whole function if that's ok

Reply:
If we are using the `get_hbond_map` function (which we do, right?) then we are covering it somewhere in the tests so please do _not_ exempt the whole function.

Only mark the small parts that are not tested, i.e., the two `else` blocks here and below. 

This provides fine-grained information once something goes wrong. If the exceptions fail somehow then we know that we should have tested. But if something else goes wrong that is actually covered then we know we have a real problem.

The alternative is writing tests to cover the raised exceptions.... excluding them is the easier path. Your choice.

Reply:
Ah, I see. Fixed then -- moved nocover to the ends of the branches (raises/returns) not covered in tests directly or indirectly.